File: Project-source/terrane-web/src/lib/branding.ts
/** Public branding (white-label) API (/api/v1/branding) — no login required, accessible while locked.
 *  Lets the frontend show the deployer's branding (logo / tab title / login subtitle) before authentication. */

import { request } from "@/lib/api";

export interface Branding {
  product_name: string;
  logo_data: string | null;     // Console / workspace logo
  login_logo: string | null;    // Login page logo
  favicon: string | null;       // Site favicon
  accent_color: string;
  login_subtitle: string | null;
  support_url: string | null;
  enabled: boolean;
}

export const DEFAULT_BRANDING: Branding = {
  product_name: "Terrane",
  logo_data: null,
  login_logo: null,
  favicon: null,
  accent_color: "#0f9b8e",
  login_subtitle: null,
  support_url: null,
  enabled: true,
};

export function getBranding(): Promise<Branding> {
  return request<Branding>("/api/v1/branding", { credentials: "include" });
}

File: Project-source/terrane-server/app/api/v1/branding.py
"""Public branding (white-label) endpoint (/api/v1/branding) — no login required, available even when locked.

The frontend logo / tab title / login-page subtitle must show the deployer's branding before authentication,
so this endpoint requires no auth and is on the license_gate allowlist (still returns when locked). Read-only;
when absent it returns the factory defaults (page-based zero-config rule).
Branding is written from the admin console (settings/wizard in terrane-admin-api).
"""

from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_db_session
from app.models.branding import SINGLETON_ID, Branding

router = APIRouter(prefix="/api/v1/branding", tags=["branding"])


@router.get("")
async def public_branding(db: AsyncSession = Depends(get_db_session)) -> dict:
    b = await db.get(Branding, SINGLETON_ID)
    return {
        "product_name": b.product_name if b else "Terrane",
        "logo_data": b.logo_data if b else None,
        "login_logo": b.login_logo if b else None,
        "favicon": b.favicon if b else None,
        "accent_color": b.accent_color if b else "#0f9b8e",
        "login_subtitle": b.login_subtitle if b else None,
        "support_url": b.support_url if b else None,
        "enabled": b.enabled if b else True,
    }
